webapp/channels/src/components/admin_console/billing/billing_subscriptions/to_paid_plan_nudge_banner.test.tsx

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {screen} from '@testing-library/react';
+
+import {renderWithIntlAndStore} from 'tests/react_testing_utils';
+import {CloudProducts} from 'utils/constants';
+
+import {ToPaidNudgeBanner, ToPaidPlanBannerDismissable} from './to_paid_plan_nudge_banner';
+
+const initialState = {
+    views: {
+        announcementBar: {
+            announcementBarState: {
+                announcementBarCount: 1,
+            },
+        },
+    },
+    entities: {
+        general: {
+            config: {
+                CWSURL: '',
+                FeatureFlagEnableCloudFreeDeprecationUI: 'true',
+            },
+            license: {
+                IsLicensed: 'true',
+                Cloud: 'true',
+            },
+        },
+        users: {
+            currentUserId: 'current_user_id',
+            profiles: {
+                current_user_id: {roles: 'system_user'},
+            },
+        },
+        preferences: {
+            myPreferences: {},
+        },
+        cloud: {},
+    },
+};
+
+describe('ToPaidPlanBannerDismissable', () => {
+    test('should only show for admins on cloud free', () => {
+        const state = JSON.parse(JSON.stringify(initialState));
+        state.entities.users.profiles = {
+            current_user_id: {roles: 'system_admin'},
+        };
+        state.entities.cloud = {
+            subscription: {
+                product_id: 'prod_starter',
+                is_free_trial: 'false',
+                trial_end_at: 1,
+            },
+            products: {
+                prod_starter: {
+                    id: 'prod_starter',
+                    sku: CloudProducts.STARTER,
+                },
+            },
+        };
+
+        renderWithIntlAndStore(<ToPaidPlanBannerDismissable/>, state);
+
+        expect(screen.getByTestId('cloud-free-deprecation-announcement-bar')).toBeInTheDocument();
```

Comment:
Nit: all the `getBy*` selectors (like `getByTestId` ) are assertions all by themselves, so we can omit the wrapping `expect(...).toBeInTheDocument()`. The test will fail if it is not found without that (as seen by the example below that expects it to throw in the not found case)